### DISCUSSION TITLE
LDS02RR Code

### DISCUSSION THREAD

**Discussion by Zando777** [makerspet/support#6] [General]
Hello, I was wondering if you could release the code that you used in the demo video of the LDS02RR (https://www.youtube.com/shorts/gaDnZ4Msw0E). I am trying to replicate what you have done on an ESP32. and maybe also a schematic of the ESP32 connections to the LDS02RR. Thank you :)

**Comment by makerspet** [comment]
Here is the code I used in that video https://github.com/kaiaai/XV_Lidar_Controller_ESP32/
Edit: this code also works https://github.com/kaiaai/LDS02RR_ESP32 It's a rewrite of the code linked above.

That code was later merged into the Arduino LDS library that supports many LiDAR types https://github.com/kaiaai/LDS

I also used these boards https://github.com/makerspet/pcb
- under esp32_breakout/output folder
- under lds02rr_adapter/output folder

**Reply by Zando777** [reply]
Thanks!

**Reply by ArtemVolkov2103** [reply]
Hello! I want to find out why the test code LDS02RR_ESP32/blob/main/lds02rr_esp32/lds02rr_esp32.ino has the pins RX and TX, although the LiDAR itself (I'm looking at makerspet/pcb/blob/main/lds02rr_adapter/output/LDS02RR_adapter_schematic.pdf) has only the TX output? Thanks!

**Comment by kaiaai** [comment]
@ArtemVolkov2103 LDS02RR replicated the Neato LiDAR protocol, but with some changes/simplifications like LDS02RR operating without the RX signal.

Initially, I got LDS02RR working by reusing the Neato XV11 code https://github.com/kaiaai/XV_Lidar_Controller_ESP32/.  Next, I rewrote/refactored that XV_Lidar_Controller_ESP32 code to look more like a software library. I kept RX - even if LDS02RR does not use it - because I want the library https://github.com/kaiaai/LDS to keep supporting both Neato XV11 and LDS02RR.

**Reply by ArtemVolkov2103** [reply]
Ok, it turns out that on the ESP32 the TX pin will simply not be used. Thanks!

**Comment by Sampolytech** [comment]
Hello, am looking for lib for LDS-007 and FM1828 LIDAR of ECOVACS
Thanks

**Reply by kaiaai** [reply]
Hello,

  I'm not aware of an LDS-007 library, but there is plenty of information on LDS-006 https://github.com/kaiaai/awesome-2d-lidars Maybe, just maybe, LDS-006 code might work with LDS-007?
